src/components/Inputs/CustomImage.js

```diff
@@ -0,0 +1,123 @@
+import { Box } from '@mui/material'
+import React, { useContext, useEffect, useState } from 'react'
+import { useRef } from 'react'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+
+const CustomImage = ({ name, value, onChange, resourceId, error, seqNo, recordId, setInitialData }) => {
+  const hiddenInputRef = useRef()
+  const { getRequest } = useContext(RequestsContext)
+
+  const [image, setImage] = useState()
+
+  useEffect(() => {
+    resourceId && getData()
+  }, [resourceId])
+
+  async function getData() {
+    const result = await getRequest({
+      extension: SystemRepository.Attachment.get,
+      parameters: `_resourceId=${resourceId}&_seqNo=${seqNo}&_recordId=${recordId}`
+    })
+    setInitialData(prevData => ({
+      ...prevData,
+      attachment: result.record
+    }))
+  }
+
+  const handleClick = () => {
+    hiddenInputRef.current.click()
+  }
+
+  const handleInputImageChange = event => {
+    const file = event?.target?.files[0]
+    if (file) {
+      const dateObject = new Date(file.lastModifiedDate)
+      const year = dateObject.getFullYear()
+      const month = dateObject.getMonth() + 1
+      const day = dateObject.getDate()
+
+      const data = {
+        resourceId: resourceId,
+        recordId: 1,
+        seqNo: 0,
+        fileName: file.name,
+        folderId: null,
+        folderName: null,
+        date: day + '/' + month + '/' + year,
+        url: null
+      }
+
+      const fileSizeInKB = Math.round(file.size / 1024)
+      if (parseInt(fileSizeInKB) > 500) {
+        alert('Allowed PNG or JPEG. Max size of 500KB.')
+
+        return
+      }
+
+      onChange(name, file)
+      onChange('attachment', data)
```

Comment:
Makes no sense to do onChange on name, file, then on attachment, data. Replace those 2 statements with onChange(data). Handle saving the data inside the form with setFieldValue